crates/symbolicator/src/services/download/mod.rs

```diff
@@ -35,20 +36,27 @@ const USER_AGENT: &str = concat!("symbolicator/", env!("CARGO_PKG_VERSION"));
 /// Errors happening while downloading from sources.
 #[derive(Debug, Error)]
 pub enum DownloadError {
-    #[error("failed to download")]
+    #[error("failed to download (IO)")]
     Io(#[source] std::io::Error),
-    #[error("failed to download")]
+    #[error("failed to download (streaming file)")]
     Reqwest(#[source] reqwest::Error),
     #[error("bad file destination")]
     BadDestination(#[source] std::io::Error),
     #[error("failed writing the downloaded file")]
     Write(#[source] std::io::Error),
     #[error("download was cancelled")]
     Canceled,
-    #[error("failed to fetch data from GCS")]
+    #[error("failed to fetch data from GCS: {0}")]
     Gcs(#[from] gcs::GcsError),
-    #[error("failed to fetch data from Sentry")]
+    #[error("failed to fetch data from Sentry: {0}")]
     Sentry(#[from] sentry::SentryError),
+    #[allow(unused)]
+    #[error("failed to fetch data from S3")]
+    S3(#[source] s3::S3Error),
```

Comment:
```suggestion
    #[error("failed to fetch data from S3: {0}")]
    S3(#[from] s3::S3Error),
```

Not sure if this makes sense? Just to be in-line with the rest of these.

Reply:
it allows me to use `.into()` instead of explicitly constructing this variant in one spot, which is nice 🎉 i'm sure it's a little debatable whether that's strictly an improvement but it follows a precedent set by other variants so i'm inclined to follow those.